src/server/engine_factoryv2.rs

```diff
@@ -0,0 +1,275 @@
+        }
+        let _kv_cfs_opts = self.inner.rocksdb_config.build_cf_opts(
+            &self.inner.block_cache,
+            self.inner.region_info_accessor.as_ref(),
+            self.inner.api_version,
+        );
+        // TODOTODO: call rust-rocks or tirocks to destroy_engine;
+        /*
+        engine_rocks::raw_util::destroy_engine(
+            tablet_path.to_str().unwrap(),
+            kv_db_opts,
+            kv_cfs_opts,
+        )?;*/
+        let _ = std::fs::remove_dir_all(tablet_path);
+        Ok(())
+    }
+}
+
+impl<ER: RaftEngine> TabletFactory<RocksEngine> for KvEngineFactoryV2<ER> {
+    fn create_tablet(&self, id: u64, suffix: u64) -> Result<RocksEngine> {
+        let mut reg = self.inner.registry.lock().unwrap();
+        if let Some(db) = reg.get(&(id, suffix)) {
+            panic!("region {} {} already exists", id, db.as_inner().path());
+        }
+
+        let kv_engine = self.create_tablet(id, suffix)?;
+        debug!("inserting tablet"; "key" => ?(id, suffix));
+        reg.insert((id, suffix), kv_engine.clone());
+        Ok(kv_engine)
+    }
+
+    fn open_tablet(&self, id: u64, suffix: u64) -> Result<RocksEngine> {
+        let mut reg = self.inner.registry.lock().unwrap();
+        if let Some(db) = reg.get(&(id, suffix)) {
+            return Ok(db.clone());
+        }
+
+        let db_path = self.tablet_path(id, suffix);
+        let db = self.open_tablet_raw(db_path.as_path(), false)?;
+        debug!("open tablet"; "key" => ?(id, suffix));
+        reg.insert((id, suffix), db.clone());
+        Ok(db)
+    }
+
+    fn open_tablet_cache(&self, id: u64, suffix: u64) -> Option<RocksEngine> {
+        let reg = self.inner.registry.lock().unwrap();
+        if let Some(db) = reg.get(&(id, suffix)) {
+            return Some(db.clone());
+        }
+        None
+    }
+
+    fn open_tablet_cache_any(&self, id: u64) -> Option<RocksEngine> {
+        let reg = self.inner.registry.lock().unwrap();
+        if let Some(k) = reg.keys().find(|k| k.0 == id) {
+            debug!("choose a random tablet"; "key" => ?k);
+            return Some(reg.get(k).unwrap().clone());
+        }
+        None
+    }
+
+    fn open_tablet_raw(&self, path: &Path, _readonly: bool) -> Result<RocksEngine> {
+        if !RocksEngine::exists(path.to_str().unwrap_or_default()) {
+            return Err(box_err!(
+                "path {} does not have db",
+                path.to_str().unwrap_or_default()
+            ));
+        }
+        let (mut tablet_id, mut tablet_suffix) = (0, 1);
+        if let Some(s) = path.file_name().map(|s| s.to_string_lossy()) {
+            let mut split = s.split('_');
+            tablet_id = split.next().and_then(|s| s.parse().ok()).unwrap_or(0);
+            tablet_suffix = split.next().and_then(|s| s.parse().ok()).unwrap_or(1);
+        }
+        self.create_tablet(tablet_id, tablet_suffix)
+    }
+
+    #[inline]
+    fn create_root_db(&self) -> Result<RocksEngine> {
```

Comment:
Perhaps we can change it to `create_share_db`. In v2, there will be no root db at all.

Reply:
share makes more sense to me as it means all regions share one kv db.